packages/trader/src/App/Components/Elements/VideoStream/video-stream.tsx

```diff
@@ -1,75 +1,279 @@
+            );
+        } else {
+            document.addEventListener(
+                'mousemove',
+                dragMoveHandler as unknown as (this: Document, ev: MouseEvent) => void
+            );
+            document.addEventListener('mouseup', dragEndHandler as unknown as (this: Document, ev: MouseEvent) => void);
+        }
+
+        return () => {
+            cancelAnimationFrame(animation_ref.current);
+            if (is_mobile) {
+                document.removeEventListener(
+                    'touchmove',
+                    dragMoveHandler as unknown as (this: Document, ev: TouchEvent) => void
+                );
+                document.removeEventListener(
+                    'touchend',
+                    dragEndHandler as unknown as (this: Document, ev: TouchEvent) => void
+                );
+                document.removeEventListener(
+                    'touchcancel',
+                    dragEndHandler as unknown as (this: Document, ev: TouchEvent) => void
+                );
+            } else {
+                document.removeEventListener(
+                    'mousemove',
+                    dragMoveHandler as unknown as (this: Document, ev: MouseEvent) => void
+                );
+                document.removeEventListener(
+                    'mouseup',
+                    dragEndHandler as unknown as (this: Document, ev: MouseEvent) => void
+                );
+            }
+
+            clearTimeout(timer_ref.current);
+        };
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, []);
 
     return (
-        <iframe
-            allow={`accelerometer; gyroscope; autoplay; encrypted-media; picture-in-picture${
-                disable_picture_in_picture ? ' "none"' : ' *'
-            };`}
-            allowFullScreen={allow_full_screen}
-            src={`https://iframe.cloudflarestream.com/${src}?${params}`}
-            data-testid={test_id}
-            width={width}
-            height={height}
-            {...props}
-        />
+        <div
+            className='player__wrapper'
+            onMouseOver={is_mobile ? undefined : () => setShowControls(true)}
+            onMouseLeave={is_mobile ? undefined : () => setShowControls(false)}
+            data-testid={data_testid}
+        >
+            <Stream
+                autoplay
+                controls={false}
+                height={is_mobile ? '184.5px' : '270px'}
+                letterboxColor='transparent'
+                loop={false}
+                preload='auto'
+                responsive={false}
+                src={src}
+                streamRef={video_ref}
+                width='100%'
+                onEnded={onEnded}
+                onLoadedMetaData={onLoadedMetadata}
+            />
+            <VideoOverlay
+                onClick={is_mobile && !is_ended ? () => setShowControls(!show_controls) : togglePlaying}
```

Comment:
On mobile onClick we will show controls (pause/play is available by clicking on the prober button), on Desktop - onClick will pause/play the video